### PR TITLE
Adding getters and setters for logger objects

### PR DESCRIPTION
**Pull request by burzum**
Added getters and setters for two classes.

### CODE REVIEW

File: src/Database/Connection.php [hunk 1]
```diff
@@ -814,13 +814,35 @@ public function logQueries($enable = null)
     public function logger($instance = null)
     {
         if ($instance === null) {
-            if ($this->_logger === null) {
```
Comment (dereuromark) [inline src/Database/Connection.php]: This is Inside the new method. Looks BC.

Comment (burzum) [inline src/Database/Connection.php]: Can I write the doc block like this?

```php
    /**
     * {@inheritDoc}
     * @@deprecated 3.5.0 Use getLogger() and setLogger() instead.
     */
```

Comment (cleptric) [inline src/Database/Connection.php]: Ahh, overlooked that 😇
@burzum minus one @ and an added blank line after `{@inheritDoc}`, yes 😄

File: src/Database/Log/LoggingStatement.php [hunk 2]
```diff
@@ -115,9 +115,30 @@ public function bindValue($column, $value, $type = 'string')
...
     {
         if ($instance === null) {
-            return $this->_logger;
+            return $this->getLogger();
         }
 
         return $this->_logger = $instance;
     }
+
+    /**
```
Comment (burzum) [inline src/Database/Log/LoggingStatement.php]: It's there? I'm blind? What exactly is it complaining about?

File: src/Database/Connection.php [hunk 3]
```diff
@@ -810,17 +810,41 @@ public function logQueries($enable = null)
...
+
+        $this->setLogger($instance);
+    }
 
-            return $this->_logger;
+    /**
+     * Sets a logger
+     *
+     * @param object $logger Logger object
+     * @return void
```
Comment (robertpustulka) [inline src/Database/Connection.php]: Can you return `$this` for method chaining?

**Review by robertpustulka — CHANGES_REQUESTED** [review]

File: src/Database/Log/LoggingStatement.php [hunk 4]
```diff
@@ -109,15 +109,37 @@ public function bindValue($column, $value, $type = 'string')
...
+     * Sets a logger
+     *
+     * @param object $logger Logger object
+     * @return void
+     */
+    public function setLogger($logger)
+    {
+        $this->_logger = $logger;
```
Comment (markstory) [inline src/Database/Log/LoggingStatement.php]: Should return $this here.

File: src/Database/Connection.php (hunk 1 above)
Comment (ravage84) [inline src/Database/Connection.php]: @burzum personally, I prefer to have blank line after inheriting, as it makes it more visible that I inherit and what I overwrite/add.

File: src/Database/Log/LoggingStatement.php [hunk 5]
```diff
@@ -109,15 +109,37 @@ public function bindValue($column, $value, $type = 'string')
      * Sets the logger object instance. When called with no arguments
      * it returns the currently setup logger instance
      *
+     * @deprecated 3.5.0 Use getLogger() and setLogger() instead.
```
Comment (ravage84) [inline src/Database/Log/LoggingStatement.php]: Can we set ``@deprecated`` to the end of the annotations?

Comment (burzum) [inline src/Database/Log/LoggingStatement.php]: It's on top in other places!?

File: src/Database/Log/LoggingStatement.php (hunk 4 above)
Comment (burzum) [inline src/Database/Log/LoggingStatement.php]: Done.

File: src/Database/Log/LoggingStatement.php (hunk 5 above)
Comment (markstory) [inline src/Database/Log/LoggingStatement.php]: Or it could not matter :smile:

File: src/Database/Log/LoggingStatement.php (hunk 4 above)
Comment (markstory) [inline src/Database/Log/LoggingStatement.php]: I don't see it changed here. I see it changed in `Connection`.

File: src/Database/Log/LoggingStatement.php [hunk 6]
```diff
@@ -109,15 +109,37 @@ public function bindValue($column, $value, $type = 'string')
...
+     */
+    public function setLogger($logger)
+    {
+        $this->_logger = $logger;
+    }
+
+    /**
+     * Gets the logger object
+     *
+     * @return object logger instance
```
Comment (robertpustulka) [inline src/Database/Log/LoggingStatement.php]: Shouln't that be `@return Cake\Database\Log\QueryLogger` ?

File: src/Database/Log/LoggingStatement.php (hunk 5 above)
Comment (dereuromark) [inline src/Database/Log/LoggingStatement.php]: As Long as we dont have a sniffer for it to automate.

Comment (burzum) [inline src/Database/Log/LoggingStatement.php]: I really don't care, my point is that it is a common PITA that one person tells you to do it this way, then somebody else comes in tells you the opposite. So I would like to get phpcs to detect it and enforce you to do it right or at least a wiki page for reference to look it up.

Comment (dereuromark) [inline src/Database/Log/LoggingStatement.php]: I agree, before there are no clear sniffers/guidelines it is not worth it.
And I agree, the order it currently is is fine.
```
@other
@param
@throws
@return
```
usually

Comment (ravage84) [inline src/Database/Log/LoggingStatement.php]: I agree, we should have a) section about the order of the doc block annotations in our [coding standards page in the CookBook](https://book.cakephp.org/3.0/en/contributing/cakephp-coding-conventions.html#commenting-code) and b) a sniff that can bug people, so I don't need to :trollface: 

@dereuromark I just searched for ``@deprecated`` in the 3.x code and my findings (samples) disagree with your order.

It seems the following is used most often (which I also remember we used normally):

````
@param
@return
@throws
@other
````

[Search on GitHub repo for "deprecated"](https://github.com/cakephp/cakephp/search?p=3&q=deprecated&type=&utf8=%E2%9C%93).

But we _could_ use the yet to be written sniff to tell us which order results in more violations in the current code and we then use the one that results in the least violations.
Though, I still prefer the ordering above, as it makes more sense to me (order of importance & of usage).

File: src/Database/Log/LoggingStatement.php (hunk 6 above)
Comment (markstory) [inline src/Database/Log/LoggingStatement.php]: I'll fix these type errors up post merge.

**Pull request merged**